libyul/AsmParser.cpp

```diff
@@ -138,28 +138,42 @@ Statement Parser::parseStatement()
 		return ExpressionStatement{locationOf(expr), expr};
 	}
 	case Token::Comma:
+	case Token::AssemblyAssign:
 	{
-		// if a comma follows, a multiple assignment is assumed
-
 		if (elementary.type() != typeid(Identifier))
-			fatalParserError("Label name / variable name must precede \",\" (multiple assignment).");
+			fatalParserError(
+				std::string("Variable name must precede ") +
+				(currentToken() == Token::Comma ? "\",\" (multiple assignment)." : "\":=\".")
+			);
+
 		Identifier const& identifier = boost::get<Identifier>(elementary);
 
+		auto checkIdentifier = [this](Identifier const& identifier)
+		{
+			if (m_dialect->builtin(identifier.name))
+				fatalParserError("Cannot assign to builtin function \"" + identifier.name.str() + "\".");
+		};
+
+		checkIdentifier(identifier);
+
 		Assignment assignment = createWithLocation<Assignment>(identifier.location);
 		assignment.variableNames.emplace_back(identifier);
```

Comment:
I still have the impression that this can be condensed even more. The initial part here is very similar to the lower part of the loop body.